### PR TITLE
Update React render functions to React 18

### PR DESCRIPTION
**Pull request by fessehaye**
# Description

Recently updated foundation to React 18. To get rid of the existing warnings, we must use the new createRoot API https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis

### CODE REVIEW

File: source/js/buyers-guide/bg-main.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ import initializeSentry from "../common/sentry-config.js";
 import PNIMobileNav from "./pni-mobile-nav.js";
 
 // Initializing component a11y browser console logging
+// TODO React-axe is currently deprecated, we should replace it with @axe-core/react
```
Comment (mmmavis) [inline source/js/buyers-guide/bg-main.js:22]: Can you create a ticket for this and update this inline comment to include link to that ticket, please?

File: source/js/nav-newsletter.js [hunk 2]
```diff
@@ -163,7 +163,8 @@ class NavNewsletter {
       props.signupId || 0
     }/`;
     props.isHidden = false;
-    this.form = ReactDOM.render(<JoinUs {...props} />, elements.joinUs);
+    const root = createRoot(elements.joinUs);
+    this.form = root.render(<JoinUs {...props} />);
```
Comment (mmmavis) [inline source/js/nav-newsletter.js:167]: I'm getting this error. See STR below

image

### STR

- click on "Newsletter" button on nav
- click on that button again to collaspe the panel
- see error

File: source/js/main.js [hunk 3]
```diff
@@ -39,6 +39,7 @@ import { initYoutubeRegretsCarousel } from "./foundation/pages/youtube-regrets/c
 import { initYoutubeRegretsLocomotiveScroll } from "./foundation/pages/youtube-regrets/locomotive-scroll";
 
 // Initializing component a11y browser console logging
+// TODO React-axe is currently deprecated, we should replace it with @axe-core/react
```
Comment (mmmavis) [inline source/js/main.js:42]: Same. Could you please update this comment with link to the to-be-created ticket?

File: source/js/foundation/inject-react/pulse-project-list.js [hunk 4]
```diff
@@ -10,7 +10,8 @@ export default (apps, env) => {
   document.querySelectorAll(`.pulse-project-list`).forEach((target) => {
     apps.push(
       new Promise((resolve) => {
-        ReactDOM.render(
+        const root = createRoot(container);
```
Comment (mmmavis) [inline source/js/foundation/inject-react/pulse-project-list.js:13]: I have question about what `container` references to

File: source/js/foundation/inject-react/pulse-tabbed-profile-filters.js [hunk 5]
```diff
@@ -36,7 +36,8 @@ export default (apps) => {
 
     apps.push(
       new Promise((resolve) => {
-        ReactDOM.render(
+        const root = createRoot(container);
```
Comment (mmmavis) [inline source/js/foundation/inject-react/pulse-tabbed-profile-filters.js:39]: Same. What is `container` referencing to?

**Review by mmmavis — CHANGES_REQUESTED** [review]
Thank you for fixing these deprecation warnings, @fessehaye !

I left some comments in this PR. Could we also replace `ReactDOM.render calls` in all inline comments to something like `React client rendering calls`?

File: source/js/foundation/inject-react/pulse-project-list.js (hunk 4 above)
Comment (fessehaye) [reply source/js/foundation/inject-react/pulse-project-list.js]: should be target!

File: source/js/foundation/inject-react/pulse-tabbed-profile-filters.js (hunk 5 above)
Comment (fessehaye) [reply source/js/foundation/inject-react/pulse-tabbed-profile-filters.js]: should also be target

File: source/js/buyers-guide/bg-main.js (hunk 1 above)
Comment (fessehaye) [reply source/js/buyers-guide/bg-main.js:22]: https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10306

**Review by mmmavis — APPROVED** [review]
🎉

**Pull request merged**